plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMergeSink.java

```diff
@@ -410,21 +414,34 @@ private Optional<DataFileInfo> rewriteParquetFile(Path path, FileDeletion deleti
                 int positionCount = page.getPositionCount();
                 int[] retained = new int[positionCount];
                 int[] deleted = new int[positionCount];
+                int[] deletedBecauseOfUpdate = new int[positionCount];
+                int[] deletedBecauseOfDelete = new int[positionCount];
                 int retainedCount = 0;
                 int deletedCount = 0;
+                int deletedBecauseOfUpdateCount = 0;
+                int deletedBecauseOfDeleteCount = 0;
                 for (int position = 0; position < positionCount; position++) {
                     if (!rowsToDelete.contains(filePosition)) {
```

Comment:
I'd rephrase these as:

```
if (rowsToDelete.contains(filePosition)) {
    ...
}
else if (rowsDeletedBecauseOfUpdate.contains(filePosition)) {
    ...
}
else {
    // retain ...
}

Reply:
But that would be incorrect. I am interested in rows that will be deleted  I just need to split them into 2 types

Reply:
Ok my bad, I figured it out and it works. Thanks